### DISCUSSION TITLE
Iterating over github.event.commits in pwsh

### DISCUSSION THREAD

**Discussion by agaertner** [community/community#35120] [Actions] [answered]
I am trying to iterate over github.event.commits in a github action script using the default shell (powershell).

I tried to convert it to json using ${{ toJSON(github.event.commits) }} then passing it to ConvertTo-Json because I thought it would result in a usable object.
I also tried to get a count by passing it to ``jq`` for a basic for loop. 
I also researched if my syntax is just wrong. My attempts were futile
Now I've read that it's basically already in json (?)  so I went with ${{ fromJSON( github.event.commits }}  but this gives me a conversion error (``Unexpected character encountered while parsing value: A. Path '', line 0, position 0.``). 

Here's the basic code without the fromJSON/toJSON futile parts.

```pwsh
run: |
      $lines = "## What’s Changed<br><br>"
      $commits = ${{ github.event.commits }}
      foreach ($commit in $commits) {
          $lines += $commit.message + " (" + $commit.id + ") @" + $commit.author.username + "<br>"
      }
      Write-Output "::set-output name=log::$lines"
```

I've been using an action for conventional changelog generation in the past, but due to my repos needs I just want to create the changelog this way now.

**Comment by agaertner** [comment, marked as answer]
Solved it by using [actions/github-script](https://github.com/marketplace/actions/github-script), putting the conversion into the environment and changing the encoding.
```xml
- name: Create Changelog
  id: changelog
  uses: actions/github-script@v6.3.1
  env:
    COMMITS: ${{ toJSON(github.event.commits) }}
  with:
    result-encoding: string
    script: |
        const commits = JSON.parse(process.env.COMMITS);
        var lines = "## What’s Changed\n\n";
        for (const commit of commits) {
          lines += "- " + commit.message + " (" + commit.id + ") @" + commit.author.username + "\n"
        }
        return lines
```
Output in ``${{ steps.changelog.outputs.result }}``

**Comment by Ribeiro-Breno** [comment]
I wanted to interate between the id's.
Here how I did

```
for commit in ${{ join(github.event.commits.*.id, ' ') }}
do
   echo commit
done
```
